servicetalk-transport-api/src/main/java/io/servicetalk/transport/api/TransportConfig.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright © 2024 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.transport.api;
+
+/**
+ * Configuration for transport settings.
+ *
+ * @see TransportConfigBuilder
+ */
+public interface TransportConfig {
```

Comment:
This config follows the patters of `[Client|Server]SslConfig`, `ProxyConfig`, `RedirectConfig`, and `[H1|H2]ProtocolConfig` by having an interface and a builder.

Reply:
For ServiceTalk API I would consider it as highly unlikely scenario that we ever change transport implementation. If we do, it will worth the major version bump and then we can also deprecate options and move them on impl-specific variants. During the transition period, it will be fine if the new impl ignores these options and logs about it or throws an exception.